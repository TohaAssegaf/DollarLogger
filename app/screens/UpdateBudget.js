import React from 'react'
import { StyleSheet, Text, View } from 'react-native'

export default class UpdateBudget extends React.Component {
  static navigationOptions = {
    header: null // Hide header bar in this screen.
  }

  render() {
    return (
      <View>
        <Text>Placeholder for UpdateBudget screen.</Text>
      </View>
    );
  }
}
