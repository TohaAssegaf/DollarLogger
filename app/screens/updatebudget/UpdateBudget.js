import * as Routes from '/app/config/Routes'
import styles from './styles'
import actions from '/app/actions'
import MoneyField from '/app/components/money/MoneyField'
import React from 'react'
import { Button, Keyboard, Text, View } from 'react-native'
import { connect } from 'react-redux'

class UpdateBudget extends React.Component {
  static navigationOptions = {
    header: null // Hide header bar in this screen.
  }

  constructor(props) {
    super(props)
    this.state = {
      total: 0
    }
  }

  updateTotal(total: number) {
    this.setState({ total })
  }

  submitForm() {
    Keyboard.dismiss()
    this.props.updateBudgetTotal(this.state.total)
    this.props.navigation.goBack()
  }

  render() {
    return (
      <View style={styles.screen}>
        <Text style={styles.header}>What is your weekly budget?</Text>
        <Text style={styles.subtitle}>(Don't worry, you can change this later)</Text>
        <MoneyField
          textInputStyles={styles.textInput}
          currencyCodeStyles={styles.currencyCode}
          onChange={(total) => this.updateTotal(total)}/>
        <View style={styles.button}>
          <View style={styles.buttonInner}>
            <Button
                onPress={() => this.submitForm()}
                title="SUBMIT"
                color={"#6A1B9A"}
                accessibilityLabel="Update weekly budget"/>
          </View>
        </View>
      </View>
    );
  }
}

const mapStateToProps = state => {
  return {
    budget: state.budget
  }
}
â€Š
const mapDispatchToProps = dispatch => {
  return {
    updateBudgetTotal: total => {
      dispatch(actions.setBudgetTotal(total))
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(UpdateBudget)
