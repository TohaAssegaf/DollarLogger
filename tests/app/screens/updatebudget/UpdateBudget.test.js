import React from 'react';
import UpdateBudget from '/app/screens/updatebudget/UpdateBudget';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const rendered = renderer.create(<UpdateBudget />).toJSON();
  expect(rendered).toMatchSnapshot();
});
