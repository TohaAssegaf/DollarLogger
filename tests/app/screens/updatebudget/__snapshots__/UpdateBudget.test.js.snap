// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#222222",
        "fontSize": 30,
      }
    }
  >
    What is your weekly budget?
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 15,
        "paddingBottom": 20,
      }
    }
  >
    (Don't worry, you can change this later)
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#222222",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 20,
          },
          Object {
            "fontSize": 30,
          },
        ]
      }
    >
      $
    </Text>
    <TextInput
      allowFontScaling={true}
      keyboardType="numeric"
      onBlur={[Function]}
      onChangeText={[Function]}
      style={
        Array [
          Object {
            "fontSize": 20,
            "minWidth": 100,
          },
          Object {
            "fontSize": 30,
          },
        ]
      }
      value={undefined}
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingTop": 40,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "paddingHorizontal": 40,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel="Update weekly budget"
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#6A1B9A",
                },
              ]
            }
          >
            SUBMIT
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
